Yosemite/Yosemite/Stores/BlazeStore.swift

```diff
@@ -216,6 +220,69 @@ private extension BlazeStore {
     }
 }
 
+// MARK: - Synchronize target topics
+private extension BlazeStore {
+    func synchronizeTargetTopics(siteID: Int64,
+                                 locale: String,
+                                 onCompletion: @escaping (Result<[BlazeTargetTopic], Error>) -> Void) {
+        Task { @MainActor in
+            do {
+                let stubbedResult = [
+                    BlazeTargetTopic(id: "IAB1", description: "Arts & Entertainment", locale: locale),
+                    BlazeTargetTopic(id: "IAB2", description: "Automotive", locale: locale)
+                ]
+                // TODO-11540: remove stubbed result when the API is ready.
+                let topics: [BlazeTargetTopic] = try await mockResponse(stubbedResult: stubbedResult, onExecution: {
+                    try await remote.fetchTargetTopics(for: siteID, locale: locale)
+                })
+                insertStoredTargetTopicsInBackground(readonlyTopics: topics, locale: locale) {
+                    onCompletion(.success(topics))
+                }
+            } catch {
+                onCompletion(.failure(error))
+            }
+        }
+    }
+
+    /// Removes BlazeTargetTopic entities with the given locale
+    /// and inserts specified BlazeTargetTopic Entities in a background thread.
+    /// `onCompletion` will be called on the main thread.
+    ///
+    func insertStoredTargetTopicsInBackground(readonlyTopics: [Networking.BlazeTargetTopic],
+                                              locale: String,
+                                              onCompletion: @escaping () -> Void) {
+        let derivedStorage = sharedDerivedStorage
+        derivedStorage.perform {
+            derivedStorage.deleteBlazeTargetTopics(locale: locale)
+            for topic in readonlyTopics {
+                let storageTopic = derivedStorage.insertNewObject(ofType: Storage.BlazeTargetTopic.self)
+                storageTopic.update(with: topic)
+            }
+        }
+
+        storageManager.saveDerivedType(derivedStorage: derivedStorage) {
+            DispatchQueue.main.async(execute: onCompletion)
+        }
+    }
+}
+
+// MARK: - Fetch target location by keyword
+private extension BlazeStore {
+    func fetchTargetLocations(siteID: Int64,
+                              query: String,
+                              locale: String,
+                              onCompletion: @escaping (Result<[BlazeTargetLocation], Error>) -> Void) {
+        Task { @MainActor in
+            do {
+                let locations = try await remote.fetchTargetLocations(for: siteID, query: query, locale: locale)
```

Comment:
❓ Should we stub response for this?

Reply:
Nice catch! Updated in e32ce05f87230eda0f2bff0fc02d569f33223d19.